File: fixtures/CsFixture/Animals.cs
using System.Collections.Generic;

namespace Fixtures
{
    // Covers: interface, method signatures, property signatures
    public interface IAnimal
    {
        string Name { get; }
        string Sound { get; }
        string Describe();
        void MakeSound();
    }

    // Covers: second interface on same type (multiple implements)
    public interface IDomesticated
    {
        string Owner { get; }
        void Domesticate(string ownerName);
    }

    // Covers: abstract class, interface implementation, fields, properties,
    //         virtual method, abstract method
    public abstract class Animal : IAnimal
    {
        private readonly string _name;
        protected string _sound;

        protected Animal(string name, string sound)
        {
            _name = name;
            _sound = sound;
        }

        public string Name => _name;
        public string Sound => _sound;

        public virtual string Describe() => $"I am {_name} and I say {_sound}";
        public abstract void MakeSound();
    }

    // Covers: : BaseClass, multiple interfaces, override, base call,
    //         object instantiation, cross-method call
    public class Dog : Animal, IDomesticated
    {
        private string _owner = string.Empty;

        public Dog(string name) : base(name, "Woof") { }

        public string Owner => _owner;

        public override void MakeSound() => Console.WriteLine(_sound);

        public override string Describe()
        {
            var baseDesc = base.Describe();
            return $"{baseDesc}, and I am a dog";
        }

        public void Domesticate(string ownerName)
        {
            _owner = ownerName;
            LogDomestication(ownerName);
        }

        private void LogDomestication(string ownerName)
        {
            Console.WriteLine($"{Name} domesticated by {ownerName}");
        }

        public void Fetch(string item)
        {
            Console.WriteLine($"{Name} fetches {item}");
        }
    }

    // Covers: single inheritance only
    public class Cat : Animal
    {
        public Cat(string name) : base(name, "Meow") { }
        public override void MakeSound() => Console.WriteLine(_sound);
    }

    // Covers: new List<>() instantiation, foreach call to interface method
    public class AnimalShelter
    {
        private readonly List<IAnimal> _animals = new List<IAnimal>();

        public void Add(IAnimal animal) => _animals.Add(animal);

        public void MakeAllSounds()
        {
            foreach (var animal in _animals)
                animal.MakeSound();
        }

        public IAnimal FindByName(string name)
        {
            var dog = new Dog(name);
            return dog;
        }

        public int Count() => _animals.Count;
    }
}

File: fixtures/VbFixture/Animals.vb
Imports System.Collections.Generic

Namespace Fixtures

    ' Covers: interface, method signatures, property signatures
    Public Interface IAnimal
        ReadOnly Property Name As String
        ReadOnly Property Sound As String
        Function Describe() As String
        Sub MakeSound()
    End Interface

    ' Covers: second interface on same type (multiple implements)
    Public Interface IDomesticated
        ReadOnly Property Owner As String
        Sub Domesticate(ownerName As String)
    End Interface

    ' Covers: MustInherit class, Implements interface, fields, properties,
    '         Overridable method, MustOverride method
    Public MustInherit Class Animal
        Implements IAnimal

        Private _name As String
        Protected _sound As String

        Public Sub New(name As String, sound As String)
            _name = name
            _sound = sound
        End Sub

        Public ReadOnly Property Name As String Implements IAnimal.Name
            Get
                Return _name
            End Get
        End Property

        Public ReadOnly Property Sound As String Implements IAnimal.Sound
            Get
                Return _sound
            End Get
        End Property

        Public Overridable Function Describe() As String Implements IAnimal.Describe
            Return String.Format("I am {0} and I say {1}", _name, _sound)
        End Function

        Public MustOverride Sub MakeSound() Implements IAnimal.MakeSound

    End Class

    ' Covers: Inherits, multiple Implements, Overrides, MyBase call,
    '         object creation (instantiates edge), cross-method call
    Public Class Dog
        Inherits Animal
        Implements IDomesticated

        Private _owner As String

        Public Sub New(name As String)
            MyBase.New(name, "Woof")
        End Sub

        Public ReadOnly Property Owner As String Implements IDomesticated.Owner
            Get
                Return _owner
            End Get
        End Property

        Public Overrides Sub MakeSound()
            Console.WriteLine(_sound)
        End Sub

        Public Overrides Function Describe() As String
            Dim base As String = MyBase.Describe()
            Return base & ", and I am a dog"
        End Function

        Public Sub Domesticate(ownerName As String) Implements IDomesticated.Domesticate
            _owner = ownerName
            LogDomestication(ownerName)
        End Sub

        Private Sub LogDomestication(ownerName As String)
            Console.WriteLine(String.Format("{0} domesticated by {1}", Name, ownerName))
        End Sub

        Public Sub Fetch(item As String)
            Console.WriteLine(String.Format("{0} fetches {1}", Name, item))
        End Sub

    End Class

    ' Covers: Inherits only (no extra interfaces)
    Public Class Cat
        Inherits Animal

        Public Sub New(name As String)
            MyBase.New(name, "Meow")
        End Sub

        Public Overrides Sub MakeSound()
            Console.WriteLine(_sound)
        End Sub

    End Class

    ' Covers: New List(Of ...) instantiation, For Each loop call,
    '         calls to interface method via variable
    Public Class AnimalShelter
        Private _animals As New List(Of IAnimal)

        Public Sub Add(animal As IAnimal)
            _animals.Add(animal)
        End Sub

        Public Sub MakeAllSounds()
            For Each animal As IAnimal In _animals
                animal.MakeSound()
            Next
        End Sub

        Public Function FindByName(name As String) As IAnimal
            Dim dog As New Dog(name)
            Return dog
        End Function

        Public Function Count() As Integer
            Return _animals.Count
        End Function

    End Class

End Namespace
